rust_snuba/rust_arroyo/src/processing/mod.rs

```diff
@@ -55,7 +65,7 @@ impl<TPayload> ConsumerState<TPayload> {
     }
 }
 
-pub struct Callbacks<TPayload>(Arc<Mutex<ConsumerState<TPayload>>>);
+pub struct Callbacks<TPayload>(pub Arc<Mutex<ConsumerState<TPayload>>>);
```

Comment:
Instead of carrying the `Arc<Mutex<ConsumerState<TPayload>>>` around in all the public APIs, how about you hide the `Arc<Mutex<>>` inside of `ConsumerState`.